darts/timeseries.py

```diff
@@ -240,10 +242,52 @@ def __init__(self, xa: xr.DataArray):
             static_covariates.columns.name = STATIC_COV_TAG
             # convert numerical columns to same dtype as series
             numeric_cols = static_covariates.select_dtypes(include=np.number).columns
-            self._xa = _xarray_with_static_covariates(
-                self._xa,
-                static_covariates.astype({col: self.dtype for col in numeric_cols}),
+            static_covariates = static_covariates.astype(
+                {col: self.dtype for col in numeric_cols}
+            )
+
+        # handle hierarchy
+        hierarchy = self._xa.attrs.get(HIERARCHY_TAG, None)
+        self._top_level_component = None
+        self._bottom_level_components = None
+        if hierarchy is not None:
+            raise_if_not(
+                isinstance(hierarchy, dict),
+                "The hierarchy must be a dict mapping (non-top) component names to their parent(s) in the hierarchy.",
+            )
+            # pre-compute grouping informations
+            components_set = set(self.components)
+            children = set().union(hierarchy.keys())
```

Comment:
Is there a specific reason to not use `set(hierarchy.keys())` ?

Reply:
`union()` accepts multiple arguments whereas `set()` only accepts one

Reply:
good point 👍